ouroboros-consensus/docs/GitProcess.md

```diff
@@ -0,0 +1,79 @@
+PR builds before it's merged, so every commit on the `master` branch will build
+and pass the tests that are enabled in CI. In particular, this process ensures
+`git bisect` can be as useful as possible -- when you reach for that kind of
+sledgehammer, you want it to work well.
+
+Note that CI enables `-Werror`, but we don't require you to enable it locally.
+If you'd like to opt-in to that or similar, add a `cabal.project.local` file.
+
+## Guidelines
+
+  * Write useful commit messages. There are lots of guides on the Internet about
+    how to do that. We don't have any strict guidelines beyond appreciating the
+    value of a good commit message and appreciating their permanence.
+
+  * We only merge via GitHub PR. Moreover, we favor the command `bors r+` or the
+    equivalent `bors merge` as the final step on a PR, when the `bors` bot is
+    behaving. Usually it's the PR author that issue that command.
+
+  * The tip of the `master` branch is always the latest and greatest we have to
+    offer, closer to our current primary goals. As mentioned above, it should
+    build clean and pass our test suites. We do also rely on some additional
+    and/or external testing that is only done near release points.
+
+  * GitHub copies the PR title and description into the merge commit, so the PR
+    title and description should be useful and meaningful. Don't use the default
+    title.
+
+    If you want to initiate discussion on the PR (eg "This is the direction I
+    come up with, what do you think?"), do so in the first comment, immediately
+    after opening the PR -- we don't want that recorded in the merge commit.
+
+  * If the PR resolves an Issue, the PR description should start with `Fixes
+    #NNNN`. See the [GitHub documentation][gh-auto-link-issue] for similar
+    keywords and syntax that will trigger useful automatic behaviors.
+
+    [gh-auto-link-issue]: https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
+
+  * Your PR should be a [Draft PR][github-draft-pr] until you think it is ready
+    for final review and merging. I often open my PR as Draft PR in order to get
+    early feedback and then later promote it to a proper PR. Or if I realize
+    there is big hole in my PR, I will convert it back into a Draft PR.
+
+    This is far superior to including "WIP" or "DO NOT MERGE" in the PR title,
+    or a WIP label, etc.
+
+    [github-draft-pr]: https://github.blog/2019-02-14-introducing-draft-pull-requests
+
+  * We're currently bridging the gap with both GitHub Issues and tickets in
+    IOHK's JIRA. For now, we are therefore also mentioning JIRA Issues such as
+    `CAD-NNNN` in PR titles and PR descriptions.
+
+  * Your branch name is impermanent, so it's less important than the above.
+    However, we prefer the format `username/issue-NNNN-short-description`.
```

Comment:
Not quite clear here on the intent. Is the `NNNN` from the JIRA's `CAD-NNNN` or the github issue's `NNNN`?

Reply:
Good catch, I'll clarify.